src/Sulu/Bundle/MediaBundle/Media/Manager/MediaManager.php

```diff
@@ -951,4 +999,56 @@ private function getNormalizedFileName($originalFileName)
 
         return $fileName;
     }
+
+    public function removeFileVersion(int $mediaId, int $version): void
+    {
+        $mediaEntity = $this->getEntityById($mediaId);
+        $file = $mediaEntity->getFiles()[0];
+
+        $currentFileVersion = null;
+
+        foreach ($file->getFileVersions() as $fileVersion) {
+            if ($fileVersion->getVersion() === $version) {
+                $currentFileVersion = $fileVersion;
+                break;
+            }
+        }
+
+        if (!$currentFileVersion) {
+            throw new NotFoundHttpException(
+                \sprintf(
+                    'Version "%s" for Media "%s" not found.',
+                    $version,
+                    $mediaId
+                )
+            );
+        }
+
+        if ($currentFileVersion === $file->getLatestFileVersion()) {
+            throw new BadRequestHttpException('Can\'t delete active version of a media.');
+        }
+
+        $this->em->remove($currentFileVersion);
+        foreach ($currentFileVersion->getFormatOptions() as $formatOptions) {
+            $this->em->remove($formatOptions);
+        }
```

Comment:
I don't know why this is necessary here, but without this, https://github.com/sulu/sulu/pull/5939/checks?check_run_id=2380363925 this error occurrs

Reply:
it is necessary because the `cascade` configurations in the media-bundle doctrine mappings are random and nobody understands them anymore 😅